mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp

```diff
@@ -1231,8 +1237,14 @@ static LogicalResult reduceMatchAndRewriteHelper(OpTy op, uint64_t axis,
       [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange blockArgs) {
         std::array<Value, 2> binaryArgs{
             blockArgs[0], isNanIgnoreMode ? blockArgs[2] : blockArgs[1]};
-        auto result = createLinalgBodyCalculationForReduceOp(
-            op, binaryArgs, elementTy, rewriter);
+
+        // If reduction type differs then extend (applicable to reduce_sum)
```

Comment:
Do you need an opType guard here if anything later mandates different handling of other reductions ?

Reply:
It is already kind of guarded by OpType as this is only true when the operation is a reduce sum. We might to rework this all-together if it mandates different handling other reductions.